sql-data-pipeline/src/main/java/io/cdap/pipeline/sql/app/core/interfaces/SQLJob.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.pipeline.sql.app.core.interfaces;
+
+import io.cdap.pipeline.sql.app.core.StructuredStatement;
+
+import java.util.List;
+
+/**
+ * Represents a SQL job to be executed.
+ */
+public interface SQLJob {
+  List<StructuredStatement> getPreActions();
```

Comment:
Originally, there was a guarantee that any pre-action would be executed before any action which must be executed before any post-action. This is necessary in cases where you have multiple sinks, and each sink has a corresponding job, but the executor must combine all jobs from sinks of the same kind (in this case, every sink runs on BigQuery so it must combine all the jobs into one job). At the moment we don't really have any particular cleanup/initialization actions that must be performed, but if we were to add view support it would require cleanup support at the very least.